src/sqlJob.ts

```diff
@@ -79,6 +80,16 @@ export class SQLJob {
    * @returns A promise that resolves to the WebSocket instance.
    */
   private getChannel(db2Server: DaemonServer): Promise<WebSocket> {
+    // Handle the scenario that server is not configured properly with full chain certificates
+    // In this scenario, the obtained CA certificate is the server certificate, not the expected root CA certificate,
+    // So the certificate verification cannot pass, should set rejectUnauthorized to false.
+    let rejectUnauthorized = true;
+    
+    if (db2Server.ca) {
```

Comment:
If we pass in the db2server.ca into the function, shouldn't the server get verified based on if it has that certificate? I'm not sure why rejectUnauthorized is not just `True` here

Reply:
Ideally it should use that specified CA certificate to verify the server's certificate. But in scenario that server does not use a self-signed certificate and is not configured properly to send the full chain certificate during the handshake, the db2server.ca is just the server certificate. Since this certificate is not a self-signed, it cannot be verified by itself. That's the reason why rejectUnauthorized needs to be false for this scenario.